ingestion/src/metadata/ingestion/source/database/dynamodb/connection.py

```diff
@@ -32,6 +33,14 @@ def get_connection(connection: DynamoDBConnection):
     return AWSClient(connection.awsConfig).get_dynamo_client()
 
 
+def check_list_tables(client):
+    """
+    Test ListTables under dynamodb
+    """
+    for _ in client.tables.all():
+        return True
```

Comment:
What are we trying to test here, if that call fails or not? Maybe something like this:
```suggestion
def check_list_tables(client) -> bool:
    """
    Test ListTables under dynamodb
    """
    try:
        client.tables.all()
    except Exception:
        return False
    return True
```

Reply:
so client.tables.all() passes even with wrong token, but when we run it as an iterator, it fails giving us a proper error.

Reply:
@ayush-shah then IMO we should have this as a comment in the method